Framework/CurveFitting/test/Functions/AsymmetricPearsonVIITest.h

```diff
@@ -0,0 +1,596 @@
+#include <memory>
+
+#include "MantidAPI/PeakFunctionIntegrator.h"
+#include "MantidCurveFitting/CostFunctions/CostFuncLeastSquares.h"
+#include "MantidCurveFitting/FuncMinimizers/LevenbergMarquardtMDMinimizer.h"
+#include "MantidCurveFitting/Functions/AsymmetricPearsonVII.h"
+#include "MantidCurveFitting/Functions/Gaussian.h"
+#include "MantidCurveFitting/Functions/Lorentzian.h"
+#include "MantidCurveFitting/Jacobian.h"
+
+using namespace Mantid;
+using namespace Mantid::API;
+using namespace Mantid::CurveFitting;
+using namespace Mantid::CurveFitting::Functions;
+using namespace Mantid::CurveFitting::CostFunctions;
+
+class AsymmetricPearsonVIITest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static AsymmetricPearsonVIITest *createSuite() { return new AsymmetricPearsonVIITest(); }
+  static void destroySuite(AsymmetricPearsonVIITest *suite) { delete suite; }
+
+  AsymmetricPearsonVIITest() : m_xValues(), m_10xValues() {
+    for (size_t i = 0; i < 2000; ++i) {
+      double x_i = -1 + 0.001 * static_cast<double>(i);
+      m_xValues.emplace_back(x_i);
+    }
+
+    for (size_t i = 0; i < 10; ++i) {
+      double x_i = -1 + 0.2 * static_cast<double>(i);
+      m_10xValues.emplace_back(x_i);
+    }
+  }
+
+  void testCategory() {
+    AsymmetricPearsonVII ap7;
+    TS_ASSERT_EQUALS(ap7.category(), "XrayDiffraction");
+  }
+
+  void testSetParameters() {
+    AsymmetricPearsonVII ap7;
+    const double ph = 0.9;
+    const double pc = 0.5;
+    const double width = 1.1;
+    const double ml = 2.0;
+    const double mr = 0.0;
+    ap7.initialize();
+    ap7.setHeight(ph);
+    ap7.setCentre(pc);
+    ap7.setFwhm(width);
+    ap7.setLeftShape(ml);
+    ap7.setRightShape(mr);
+
+    TS_ASSERT_DELTA(ap7.height(), ph, 1.0e-6);
+    TS_ASSERT_DELTA(ap7.centre(), pc, 1.0e-6);
+    TS_ASSERT_DELTA(ap7.fwhm(), width, 1.0e-6);
+    TS_ASSERT_DELTA(ap7.leftShape(), ml, 1.0e-6);
+    TS_ASSERT_DELTA(ap7.rightShape(), mr, 1.0e-6);
+  }
+
+  /** Test expected output
+   * @brief testExpectedOutput: test if the asymmetric Pearson VII
+   * function returns the expected values on the interval [-1, 1)
+   * and step 0.2
+   */
+  void testExpectedOutput() {
+    const double ph = 10.0;
+    const double pc = 0.0;
+    const double width = 0.1;
+    const double ml = 10.0;
+    const double mr = 1.0;
+
+    IPeakFunction_sptr ap7 = getInitializedAPVII(ph, pc, width, ml, mr);
+    FunctionDomain1DVector domain(m_10xValues);
+    FunctionValues values_ap7(domain);
+    ap7->function(domain, values_ap7);
+
+    double expected_output[10] = {1.86665e-14, 1.34216e-12, 2.85522e-10, 3.33551e-7, 0.00477417,
+                                  10.,         0.588235,    0.153846,    0.0689655,  0.0389105};
+
+    for (size_t i = 0; i < values_ap7.size(); ++i) {
+      TS_ASSERT_DELTA(expected_output[i], values_ap7[i], 1e-6);
```

Comment:
you should use different input parameters. 40% of your expected output is "0" as the value is smaller than the delta

Reply:
Great point, thanks! I've change the input.